docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+- on-chain mempool with bidding
+- quic adaption for tpu (for ipv6-mtu-exeeding transaction packet and ddos prevention)
+
+### Scrambling details
+
+As said above, scrambling refers to a specially-arranged encryption scheme where
+anyone can decrypt (= reveal) the transaction details after a non-parallelizable
+computation which takes certain amount of time.
+
+On top of it, scrambling also has a nice property where the revelation can
+easily be verified by anyone with some kind of message authentication tag. This
+means rpc node doesn't need GPU.
+
+All of fields in a transaction are scrambled with randomized padding, including
+fee payer to avoid any kind of fingerprinting while allowing garbage-based
+spamming with amplification risk.
+
+Also, scrambled transaction expires very quickly based on wall clock time, less
+than the targeted duration of difficulty to protect from processing stalls by
+malicious leaders for MEV.
+
+So, scrambled transaction binary format will be like this
+
+```
+[scrambled transaction marker]
+[seed to recursive sha256]
+[recursion count]
+aes256-ed..
+  [aes256 key used to scramble] # this works as authentication tag
+  [expiration in unix time]
+  [normal transaction details...] # fee payer sig cover the scrambling key to avoid unauthorized scrambling
+```
+
+So, this assumes quic to bypass the transaction udp packet size limit.
+
+The number of sha256 recursion is chosen so that anyone can't reveal the
+transaction details, until the cluster optimistically commits the ordering of
+transactions in a block with very high certainty. So, this assumes the recursive
+sha256 is exhibits rather uniform throughput across various hardware (cpu/gpu)
+on the market.
+
+For instance, difficulty can be adjusted 6 9s of availability of MEV-immunity
+status based on standard deviation of past conf time. or hard-coded initially.
+
+Scrambling is optional. While execution of not-scrambled transaction is stalled
+still, infallible and/or MEV-free simple transactions can opt not to be
+scrambled because scrambling might not be desirable for all of situations (HD
+wallets and program deploy transactions).
+
+### Scrambling preparation by clients
+
+Clients need to prepare valid scrambling challenge secrets. By definition, they
+have no choice but to compute the recursive sha256, starting from
+locally-available secure random as seed. However, the challenge secrets are
+designed to be pre-computed before the actual transaction construction/signing.
+So, they can prepare those challenge secrets in background and idling manner for
+later use.
+
+### 2 step voting and voting lock-out
+
+pre-vote and vote are expected to be coalesced to avoid doubling the consensus
+communication overhead. This means a single voting transaction will include
+instructions to pre-vote on block N and to vote on block N-1. This shouldn't
+badly affect latency because block time is still 400ms frequency. Also, the
+pre-vote instruction must contain timestamp.
```

Comment:
> Also, the pre-vote instruction must contain timestamp.

Could you clarify why it needs to contain timestamp and who's local clock will be used in that case ... can't we rely on block number to determine how much time has passed ?

Reply:
now that, i think this prevote thing can be entirely removed...

that said, the timestamp is needed for more granular expiration of scrambled tx. currently, block is 400ms and sysvar clock is 1s....

Reply:
> now that, i think this prevote thing can be entirely removed...

could you expand ?